### PR TITLE
118 orthanc reciever agent to download stable studies

### LINKED ISSUE #24: Example medical imaging product
## Description

As a developer, <br>
So that I can make example jobs for my project, <br>
I want a pipeline that takes input and creates a simulated medical image output

## Acceptance Criteria
_What are some implicit tasks/criteria to be achieved within this story?_
- [ ] _Pipeline takes in necessary input for a medical imaging product that would simulate a real world job
- [ ] _Outputs a mock medical image product that would emulate jobs that our enterprise tool would handle

### LINKED ISSUE #34: Hospital viewing Analytics
## Description
_Use the following format to name and describe your user stories_

As a "hospital", <br>
So that "I can evaluate the product", <br>
I want "to metrics from past studies"

_Example:_
_As a customer,_
_so that I can see my past purchases,_
_I want to view my billing history_

## Acceptance Criteria
_What are some implicit tasks/criteria to be achieved within this story?_
- [ ] _Criteria/Task 1 User (hospital) can see analytics through AI

### LINKED ISSUE #35: Hospital Viewing Results
## Description
_Use the following format to name and describe your user stories_

As a "hospital", <br>
So that "I can see the results", <br>
I want "the results to be compatible with DICOM standard"

_Example:_
_As a customer,_
_so that I can see my past purchases,_
_I want to view my billing history_

## Acceptance Criteria
_What are some implicit tasks/criteria to be achieved within this story?_
- [ ] _Criteria/Task 1 User (hospital) can see results in their PACS system
- [ ] _Criteria/Task 2 Deliverables are correctly written out in DICOM format

### LINKED ISSUE #61: Orthanc Setup
## Description
As a hospital administrator, <br>
So that we can efficiently manage patient data and collaborate with external systems, <br>
I want to upload DICOM images to Orthanc and send them to our "Example tool pipeline".

## Acceptance Criteria
- [x] Hospital staff can upload DICOM images to Orthanc.
- [x] Orthanc securely stores uploaded DICOM images and associated metadata.
- [x] Orthanc provides a seamless integration with the hospital's existing workflow for uploading DICOM images.
- [x] Uploaded DICOM images are transmitted from Orthanc to the "Example tool pipeline" for further processing.
- [x] Orthanc ensures the integrity and confidentiality of DICOM image data during transmission to the "Example tool pipeline".
- [x] Upon successful transmission, Orthanc provides confirmation to hospital staff that the DICOM images have been sent to the "Example tool pipeline".
- [x] Orthanc logs all data transmission activities for auditing and troubleshooting purposes.

### LINKED ISSUE #68: Example Tool Integration
## Description

As a developer, <br>
So that I can ensure works correctly, <br>
I want to use example medical tool for job tracking within the system.

## Acceptance Criteria
- [ ] Extend Example Medical Tool to use compatible logging
- [ ] Run the example tool within the system
- [ ] ensure the log files can correctly update the database
- [ ] confirm the produced logs successfully display on the front end

### PR DESCRIPTION
**Pull request by mbrzus**
# Overview

This pull request introduces the fully functional system orchestrating the orthanc PACS servers with the example medical imaging product and connected to the TrackerAPI system.

This includes:
1. our internal Orthanc PACS server
2. "fake" hospital PACS server
3.  Orthanc Receiver Agent: an agent listening for new connections to our internal server. The agent manages the full data lifecycle within our system:
    - data receiving managment
    - data download to local system
    - data processing calling the example product
    - data upload to internal PACS
    - data return to the hospital
    - data deletion
4. Othanc Logger: logs all study events to trackerAPI system

# Testing
in your local environment:
1. Start your virtual environments
2. Start the PACS servers and upload example data
`bash example_tool/Orthanc/tmux_startup.sh`
3. Start the receiver agent
`python3 internal_servers/orthanc_reciever_agent.py`
4. Send data to modality in example hospital orthanc (manual click on the website)
5. The agent will perform all the work automatically

Closes #24 
Closes #34
Closes #35 
Closes #61 
Closes #68

### CODE REVIEW

File: internal_servers/orthanc_reciever_agent.py [hunk 1]
```diff
@@ -125,11 +134,38 @@ def process_data(brains_tool_path: str, study_id: str, input_data_path: str, out
         print(f"Error executing script: {e}")
 
 
-def send_data_to_hospital(data_path, hospital_id):
+def upload_data_to_internal(directory_path: str, orthanc: pyorthanc.Orthanc):
```
Comment (zmorris248) [inline internal_servers/orthanc_reciever_agent.py:141]: What is the difference between upload_data_to_internal() and return_to_original_PACS()? What is each used for?

Comment (mbrzus) [reply internal_servers/orthanc_reciever_agent.py:141]: I will provide a bit longer response to make sure there is enough context.

!!! IMPORTANT: only PACS system can communicate with PACS system !!!

This is the simplified workflow:
1. Hospital PACS sends data to our internal PACS
2. Internal downloads data to local system
3. data is processed
4. new data is uploaded back to the Internal PACS
5. Internal PACS returns data back to the Hospital PACS

We cannot return the data directly to the hospital. We first have to upload the results to the internal PACS and then only the PACS can send data back to the Hospital PACS

Let me know if there are other questions about that.

**Comment by mbrzus** [comment]
> This looks really good! I think it needs to be rebased with main and made sure that the orthanc receiving and logging agents have the updated functionality and retested

I rebased with main.

Additionally, we might want to rename parts of the codebase from job to study. For example I still see the class from trackerAPI called **JobConfigManager**

**Review by Ivan-E-Johnson — APPROVED** [review]
Looks Good

**Pull request merged**